lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -518,6 +520,30 @@ data StakeDistribution = StakeDistribution
     , unassigned :: Quantity "lovelace" Word64
     } deriving (Eq, Show, Generic)
 
+-- | Information about a stake pool. This information is not used directly by
+-- cardano-wallet. It is sourced from the stake pool registry and passed
+-- straight through to API consumers.
+data StakePoolMetadata = StakePoolMetadata
+    { ticker :: StakePoolTicker
+    -- ^ Short human-readable ID for the stake pool.
+    , homepage :: Text
+    -- ^ Absolute URL for the stake pool's homepage link.
+    , pledgeAddress :: Text
+    -- ^ Bech32-encoded address.
```

Comment:
**[question]**
I wonder if we should keep a decoded representation here as an `Address`? That's a lot of encoding/decoding done for nothing however.

Reply:
Exactly - there is no need to decode then re-encode the address.